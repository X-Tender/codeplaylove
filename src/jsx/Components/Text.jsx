import React from 'react';
import PropTypes from 'prop-types';

const Text = ({ children, className, twoColumns }) => {
	const classes = `text ${twoColumns ? 'text--two-columns' : ''} ${className}`;
	return <div className={classes}>{children}</div>;
};

Text.propTypes = {
	children: PropTypes.node,
	className: PropTypes.string,
	twoColumns: PropTypes.bool,
};

Text.defaultProps = {
	children: null,
	className: '',
	twoColumns: false,
};

export default Text;
