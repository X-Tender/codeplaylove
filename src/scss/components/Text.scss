.text {
	grid-column: 1 / -1;

	&--two-columns {
		@include md {
			column-count: 2;
			column-gap: 25px;
		}
	}
}
