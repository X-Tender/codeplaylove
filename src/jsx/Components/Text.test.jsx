import React from 'react';
import { shallow } from 'enzyme';
import Text from './Text';

describe('Text', () => {
	const className = 'text-class';
	const children = <p>Lorem Ipsum dolor sit ammet</p>;

	describe('Simple Text', () => {
		const simpleText = shallow(<Text>{children}</Text>);
		it('renders correctly', () => {
			expect(simpleText).toMatchSnapshot();
		});

		it('renders correct props', () => {
			expect(
				simpleText
					.find('div.text')
					.childAt(0)
					.equals(children)
			).toBe(true);

			expect(simpleText.find('div.text').hasClass('text--two-columns')).toBe(false);
		});
	});

	describe('Text with class', () => {
		const textWithClass = shallow(<Text className={className}>{children}</Text>);

		it('renders correctly', () => {
			expect(textWithClass).toMatchSnapshot();
		});

		it('renders correct props', () => {
			expect(
				textWithClass
					.find('div.text')
					.childAt(0)
					.equals(children)
			).toBe(true);

			expect(textWithClass.find(`div.text.${className}`).exists()).toBe(true);

			expect(textWithClass.find('div.text').hasClass('text--two-columns')).toBe(false);
		});
	});

	describe('Two column text', () => {
		const twoColumnText = shallow(<Text hasTwoColumns>{children}</Text>);

		it('renders correctly', () => {
			expect(twoColumnText).toMatchSnapshot();
		});

		it('renders correct props', () => {
			expect(
				twoColumnText
					.find('div.text')
					.childAt(0)
					.equals(children)
			).toBe(true);

			expect(twoColumnText.find('div.text').hasClass('text--two-columns')).toBe(true);
		});
	});
});
