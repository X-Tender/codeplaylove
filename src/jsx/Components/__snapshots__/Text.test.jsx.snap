// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Simple Text renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Text
    className=""
    hasTwoColumns={false}
  >
    <p>
      Lorem Ipsum dolor sit ammet
    </p>
  </Text>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Lorem Ipsum dolor sit ammet
      </p>,
      "className": "text  ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem Ipsum dolor sit ammet",
      },
      "ref": null,
      "rendered": "Lorem Ipsum dolor sit ammet",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Lorem Ipsum dolor sit ammet
        </p>,
        "className": "text  ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem Ipsum dolor sit ammet",
        },
        "ref": null,
        "rendered": "Lorem Ipsum dolor sit ammet",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Text Text with class renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Text
    className="text-class"
    hasTwoColumns={false}
  >
    <p>
      Lorem Ipsum dolor sit ammet
    </p>
  </Text>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Lorem Ipsum dolor sit ammet
      </p>,
      "className": "text  text-class",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem Ipsum dolor sit ammet",
      },
      "ref": null,
      "rendered": "Lorem Ipsum dolor sit ammet",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Lorem Ipsum dolor sit ammet
        </p>,
        "className": "text  text-class",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem Ipsum dolor sit ammet",
        },
        "ref": null,
        "rendered": "Lorem Ipsum dolor sit ammet",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Text Two column text renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Text
    className=""
    hasTwoColumns={true}
  >
    <p>
      Lorem Ipsum dolor sit ammet
    </p>
  </Text>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Lorem Ipsum dolor sit ammet
      </p>,
      "className": "text text--two-columns ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem Ipsum dolor sit ammet",
      },
      "ref": null,
      "rendered": "Lorem Ipsum dolor sit ammet",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Lorem Ipsum dolor sit ammet
        </p>,
        "className": "text text--two-columns ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem Ipsum dolor sit ammet",
        },
        "ref": null,
        "rendered": "Lorem Ipsum dolor sit ammet",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
